server/rpm/pbench-server.spec

```diff
@@ -0,0 +1,241 @@
+Name:           pbench-server
+Version:        
+%define gdist   none
+Release:        1
+Summary:        The pbench server bits
+
+License:        GPLv3+
+URL:            http://github.com/distributed-systems-analysis/pbench
+Source0:        pbench-server-%{version}.tar.gz
+Buildarch:      noarch
+
+# policycoreutils for semanage and restorecon - used in pbench-server-activate-create-results-dir
+Requires:       policycoreutils
+
+%if 0%{?fedora}
+Requires:  policycoreutils-python-utils, python3-boto3
+%endif
+
+%if 0%{?rhel} == 7
+# need python3 - install scl to get it
+# scl-utils is in the main repo, but the scl python3
+# collection du jour is in the scl repo which
+# is assumed to have been added already with:
+#    yum install yum-utils
+#    yum-config-manager --enable rhel-server-rhscl-7-rpms
+#
+# we can't do that in the spec file, because of the yum lock,
+# so they have to be done beforehand.
+Requires: scl-utils, rh-python36, policycoreutils-python
+%endif
+
+%if 0%{?rhel} == 8
+# no python3-elasticsearch??
+# Requires:  policycoreutils-python-utils, python3-elasticsearch
+Requires:  policycoreutils-python-utils
+%endif
+
+Requires: npm
+
+# installdir has to agree with the definition of install-dir in
+# pbench-server.cfg, but we can't go out and pluck it from there,
+# because we don't know where the config file is. Note that we omit
+# the initial / - it is added in every use below.  IMO, that's more
+# readable since it appear in the middle of the path in all cases,
+# *except* in the %files section (and one instance in the %post
+# and %postun sections).
+
+%define installdir opt/pbench-server
```

Comment:
Should we just document that the pbench-server (and for that matter, the pbench-agent) RPM installs have the spec file hard-coded to be `/opt/pbench-server`?  If so, then we DO know where the configuration file is, `/opt/pbench-server/lib/config/pbench-server.cfg`.

Reply:
Six-of-one ... I'm changing my mind about it every time you raise the issue. Although this is the default, we *can* change it on the command line, by modifying the `rpmbuild` invocation in the Makefile to `rpmbuild --define 'installdir foo/bar'`, and that *might* be useful when testing; practically speaking we have never availed ourselves of this capability. The ansible playbooks alsio set the default value  to  `/opt/pbench-server`,  but it too can be overridden (IIRC) e.g. in the inventory. So if we want to change it, we need to change it in two places.

So we can certainly say in the doc that the installation directory is `/opt/pbench-server` and say nothing about changing it. Or we can say the default installation directory is `/opt/pbench-server` and tell how to change it if necessary. Either is fine, but it does not imply any code changes - right?

Reply:
Seems like the answer is to modify the ansible playbooks to assume (and verify) that the pbench-server RPM installation will always be in `/opt/pbench-server`, flagging a problem when the `install-dir` prefix does not match.

Another option is to remove `install-dir` from the pbench-server.cfg file, and derive the `install-dir` value in the ansible playbooks from an environment variable.  Then have the environment variable set and recorded by the RPM installation.

Regardless, seems like we should not tackle this in this PR.  Perhaps you could open an issue to track for the future?

Reply:
Filed #1803 to track.